cache/lru_cache.cc

```diff
@@ -753,7 +754,8 @@ void LRUCache::WaitAll(std::vector<Handle*>& handles) {
       }
       uint32_t hash = GetHash(handle);
       LRUCacheShard* shard = static_cast<LRUCacheShard*>(GetShard(Shard(hash)));
-      shard->Promote(lru_handle);
+      // Since wait is needed, we assume these handles are still in sec cache.
```

Comment:
This assumption works for now, but I don't think we should depend on it. I thought the previous approach of communicating it in the SecondaryCacheResultHandle was good. But if that's not preferred, then perhaps we can say that the secondary cache should return its intent to erase in Lookup(). That decision can most likely be made using some metadata in memory, rather than waiting for the entire item to be read into memory.

Reply:
This can be improved in the PR to prevent recompressing cold block since we will have some signals about whether the handle is still in the secondary cache.

Reply:
See comment on line 469.